hail/python/hailtop/hailctl/hdinsight/start.py

```diff
@@ -0,0 +1,156 @@
+    exec('az', 'hdinsight', 'script-action', 'execute', '-g', args.resource_group, '-n', 'installhail',
+         '--cluster-name', args.cluster_name,
+         '--script-uri', args.install_hail_uri,
+         '--roles', 'headnode', 'workernode',
+         '--persist-on-success',
+         '--script-parameters', f'{args.wheel_uri} {wheel_pip_version} {args.cluster_name}')
+
+    print(f'Installing Hail\'s native dependencies on {args.cluster_name}')
+    exec('az', 'hdinsight', 'script-action', 'execute', '-g', args.resource_group, '-n', 'installnativedeps',
+         '--cluster-name', args.cluster_name,
+         '--script-uri', args.install_native_deps_uri,
+         '--roles', 'headnode', 'workernode',
+         '--persist-on-success')
+
+    if args.vep:
+        if args.vep == 'GRCh38':
+            image = 'konradjk/vep95_loftee:0.2'
+        elif args.vep == 'GRCh37':
+            image = 'konradjk/vep85_loftee:1.0.3'
+        else:
+            print(f'unknown reference genome {args.vep}')
+            sys.exit(1)
+
+        vep_config_uri = args.vep_config_uri
+        if vep_config_uri is None:
+            vep_config_uri = f'https://raw.githubusercontent.com/hail-is/hail/{pip_version()}/hail/python/hailtop/hailctl/hdinsight/resources/vep-{args.vep}.json'
+
+        print(f'Loading VEP into ABS container {args.cluster_name}')
+        for uri in [args.vep_loftee_uri, args.vep_homo_sapiens_uri]:
+            exec('az', 'storage', 'copy', '--recursive', '--source', uri, '--destination', f'https://{args.storage_account}.blob.core.windows.net/{args.cluster_name}/')
+
+        print(f'Installing VEP on {args.cluster_name}')
+        exec('az', 'hdinsight', 'script-action', 'execute', '-g', args.resource_group, '-n', 'installvep',
+             '--cluster-name', args.cluster_name,
+             '--script-uri', args.install_vep_uri,
+             '--roles', 'headnode', 'workernode',
+             '--persist-on-success',
+             '--script-parameters', f'/{os.path.basename(args.vep_loftee_uri)} /{os.path.basename(args.vep_homo_sapiens_uri)} {args.vep} {image} {vep_config_uri}')
+
+    stop = json.dumps({
+        "RequestInfo": {"context": "put services into STOPPED state"},
+        "Body": {"ServiceInfo": {"state" : "INSTALLED"}}
+    })
+    start = json.dumps({
+        "RequestInfo": {"context": "put services into STARTED state"},
+        "Body": {"ServiceInfo": {"state" : "STARTED"}}
+    })
+    stop_curl, start_curl = [f'''curl -u admin:{shq(http_password)} \
+    -H 'X-Requested-By: ambari' \
+    -X PUT \
+    -d {shq(command)} https://{shq(args.cluster_name)}.azurehdinsight.net/api/v1/clusters/{shq(args.cluster_name)}/services/JUPYTER/'''
+                             for command in [stop, start]]
+
+    print(f'!!!! Restarting Jupyter. When prompted about the host key, type yes. When prompted for a password type {sshuser_password}.')
```

Comment:
so we just create a single tunnel for jupyter, yeah? Is it possible to see the spark webui in the browser, or just through the widget?

Reply:
There's actually no tunnel. I couldn't figure out how to restart Jupyter without ssh'ing to the head node and hitting that HTTP endpoint (from within the cluster network).

Azure handles all the forwarding via this `CLUTER_NAME.azurehdinsight.net`. The Spark history page, jupyter, and a few other services that we don't use are all accessible via that domain, if you have the http username and password.

Reply:
Hmm. This should totally not need to be from inside the head node. Let me try this again now.

Reply:
Yep, I can do this entirely on the public internet. Awesome, no SSH at all.